propel/internal/utils/utils.go

```diff
@@ -5,13 +5,31 @@ import (
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/meta"
 
+	client "github.com/propeldata/terraform-provider-propel/propel_client"
 	"github.com/propeldata/terraform-provider-propel/version"
 )
 
+var (
+	dataSourceTypeMap = map[client.DataSourceType]string{
+		client.DataSourceTypeSnowflake: "Snowflake",
+		client.DataSourceTypeS3:        "S3",
+		client.DataSourceTypeHttp:      "Http",
+		client.DataSourceTypeWebhook:   "Webhook",
+	}
+)
+
 // GetUserAgent augments the default user agent with provider details
 func GetUserAgent(clientUserAgent string) string {
 	return fmt.Sprintf("terraform-provider-propel/%s (terraform %s) %s",
 		version.ProviderVersion,
 		meta.SDKVersionString(),
 		clientUserAgent)
 }
+
+func GetDataSourceType(dataSourceType client.DataSourceType) string {
+	if dsType, ok := dataSourceTypeMap[dataSourceType]; ok {
+		return dsType
+	}
+
+	return ""
```

Comment:
Is this actually necessary? Could we just stringify the `dataSourceType`?

Reply:
necessary yes, because we are using the Data Source types in Terraform as camel case, but the GraphQL schema response some of them are in uppercase like `CLICKHOUSE` or `WEBHOOK` and others like `Snowflake` just capitalize the first letter.
 For the ClickHouse case, we want to use it in camel case `ClickHouse`, not `Clickhouse`